### PR TITLE
Check labels in setitem (issue 269)

### PR DESCRIPTION
**Pull request by alixdamman**
will be rebased after merge of PR #491.

### CODE REVIEW

**Comment by alixdamman** [comment]
That was the idea :)

**Comment by gdementen** [comment]
please squash & merge

**Comment by alixdamman** [comment]
OK for the name of the section for the added log?

**Comment by gdementen** [comment]
Yes, that's fine. Now that I think of it, we need one (or two) example(s) in the changelog on how to get the previous behaviour (using drop_labels/set_labels/set_axes)

File: doc/source/changes/version_0_27.rst.inc [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    >>> arr
+    a  a0  a1  a2  a3
+        0   1   2   3
+    >>> arr['a0,a1'] = arr['a2,a3']
+    ValueError: incompatible axes:
+    Axis(['a0', 'a1'], 'a')
+    vs
+    Axis(['a2', 'a3'], 'a')
+
+  previous behavior can be recovered through `drop_labels` or may change labels via
```
Comment (gdementen) [inline doc/source/changes/version_0_27.rst.inc]: s/may change/by changing/

**Comment by gdementen** [comment]
why did you remove this from the 0.27 milestone?

**Comment by alixdamman** [comment]
Because it's a PR

**Pull request merged**